### PR TITLE
add v2ray formula

### PR DESCRIPTION
**Pull request by khsing**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/v2ray.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    url "https://github.com/v2ray/geoip/releases/download/202001070102/geoip.dat"
+    sha256 "434d3f415655baa2e991a78786defe5dff798f245445cfe4580ac5d6a4a3eecf"
+  end
+
+  def install
+    ENV.append "CGO_ENABLED", 0
+    ldflags = "-s -w -X v2ray.com/core.codename=homebrew -X v2ray.com/core.build=#{DateTime.now.strftime("%Y%m%d-%H%M%S")} -X v2ray.com/core.version=#{version}"
+    system "go", "build", "-o", "#{bin}/#{name}", "-ldflags", "'#{ldflags}'", "./main"
+    system "go", "build", "-o", "#{bin}/v2ctl", "-ldflags", "'#{ldflags}'", "./infra/control/main"
+    rm  "release/config/geoip.dat"
```
Comment (fxcoudert) [inline Formula/v2ray.rb]: Why removing default configuration files?

File: Formula/v2ray.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+class V2ray < Formula
+  desc "Project V provide a unified platform for anti-censorship"
+  homepage "https://github.com/v2ray/v2ray-core"
+  version "4.22.1"
```
Comment (fxcoudert) [inline Formula/v2ray.rb]: Remove `version`, and replace it in the `url` below

File: Formula/v2ray.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      <key>WorkingDirectory</key>
+      <string>/tmp</string>
+    </dict>
+    </plist>
+  EOS
+  end
+
+  test do
+    require "open3"
+    Open3.popen3("#{bin}/v2ray", "-test -config #{etc}/v2ray/config.json") do |_, stdout, _|
```
Comment (fxcoudert) [inline Formula/v2ray.rb]: Use `shell_output` instead

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/v2ray.rb (hunk 1 above)
Comment (khsing) [reply Formula/v2ray.rb]: this config is not the latest and can't start service by default

File: Formula/v2ray.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  depends_on "geoip"
+
+  resource "geosite" do
+    url "https://github.com/v2ray/domain-list-community/releases/download/202001070832/dlc.dat"
+    sha256 "f58fa04f8c943584a3b3ffc5678f1dc27abc429a9ff600868ba97f313938f2e9"
+  end
+  resource "geoip" do
+    url "https://github.com/v2ray/geoip/releases/download/202001070102/geoip.dat"
+    sha256 "434d3f415655baa2e991a78786defe5dff798f245445cfe4580ac5d6a4a3eecf"
+  end
```
Comment (SMillerDev) [inline Formula/v2ray.rb]: Is this different from the geoip dependency above?

File: Formula/v2ray.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/v2ray.rb]: If the default config is wrong this should be reported and fixed upstream.

File: Formula/v2ray.rb (hunk 4 above)
Comment (khsing) [reply Formula/v2ray.rb]: I just removed GeoIP dependency,  v2ray has an ASSET directory setting, geoip.dat, and geosite.dat   both need put in the directory.  If depends GeoIP, may need additional symbol link to config dir. I think it's not good.

**Comment by chenrui333** [comment]
@BrewTestBot test this please

**Pull request closed without merging**